pinot-query-planner/src/test/resources/queries/WindowFunctionPlans.json

```diff
@@ -1107,6 +1323,22 @@
           "\n"
         ]
```

Comment:
Can we also cover the subquery style in the plans ?

Something like 

```
SELECT * FROM (
    SELECT 
        ROW_NUMBER() OVER (.....) rn, 
        c1, 
        c2, 
        c3
    FROM
        foo
    ) temp    
WHERE
    rn > 10 AND rn <=20;
```

Reply:
@siddharthteotia I actually have a couple of plans of these types at the end of this file. Are these sufficient?:

```
      {
        "description": "Window function CTE: row_number WITH statement having OVER with PARTITION BY ORDER BY",
        "sql": "EXPLAIN PLAN FOR WITH windowfunc AS (SELECT a.col1, ROW_NUMBER() OVER(PARTITION BY a.col2 ORDER BY a.col3) as rownum from a) SELECT a.col1, a.rownum FROM windowfunc AS a where a.rownum < 5",
        "output": [
          "Execution Plan",
          "\nLogicalProject(col1=[$0], $1=[$3])",
          "\n  LogicalFilter(condition=[<($3, 5)])",
          "\n    LogicalWindow(window#0=[window(partition {1} order by [2] rows between UNBOUNDED PRECEDING and CURRENT ROW aggs [ROW_NUMBER()])])",
          "\n      PinotLogicalSortExchange(distribution=[hash[1]], collation=[[2]], isSortOnSender=[false], isSortOnReceiver=[true])",
          "\n        LogicalProject(col1=[$0], col2=[$1], col3=[$2])",
          "\n          LogicalTableScan(table=[[a]])",
          "\n"
        ]
      },
      {
        "description": "Window function subquery: row_number having OVER with PARTITION BY ORDER BY",
        "sql": "EXPLAIN PLAN FOR SELECT row_number, col2, col3 FROM (SELECT ROW_NUMBER() OVER(PARTITION BY a.col2 ORDER BY a.col3 DESC) as row_number, a.col2, a.col3 FROM a) WHERE row_number <= 10",
        "output": [
          "Execution Plan",
          "\nLogicalProject(row_number=[$2], col2=[$0], col3=[$1])",
          "\n  LogicalFilter(condition=[<=($2, 10)])",
          "\n    LogicalWindow(window#0=[window(partition {0} order by [1 DESC] rows between UNBOUNDED PRECEDING and CURRENT ROW aggs [ROW_NUMBER()])])",
          "\n      PinotLogicalSortExchange(distribution=[hash[0]], collation=[[1 DESC]], isSortOnSender=[false], isSortOnReceiver=[true])",
          "\n        LogicalProject(col2=[$1], col3=[$2])",
          "\n          LogicalTableScan(table=[[a]])",
          "\n"
        ]
      }
```

Reply:
Looks good. Thanks for clarifying

Can we add an example of a JOIN query with ROW_NUMBER where the subquery does a JOIN and we pick the particular ROW_NUMBER for each partition in the outer query ?

```
SELECT 
<columnList>
FROM 
   (
        SELECT
        <columnList>,
        ROW_NUMBER OVER(PARTITION BY .. ORDER BY ... DESC) AS rn
        FROM 
        T1 INNER JOIN T2
        ON T1.c1 = T2.c2
        WHERE .......
  )
WHERE rn = 1
```

So we pick the highest guy (since each partition is ordered descending) in the outer query from each partition.

Reply:
done as part of https://github.com/apache/pinot/pull/10684